### PR TITLE
pandas 2.0 removals in testing and clipboard

### PR DESCRIPTION
**Pull request by hamdanal**
- [x] Work on #624
- [ ] Tests added

Reference: https://pandas.pydata.org/docs/dev/whatsnew/v2.0.0.html#removal-of-prior-version-deprecations-changes

### CODE REVIEW

File: pandas-stubs/_testing/__init__.pyi [hunk 1]
```diff
@@ -48,11 +42,7 @@ def assert_numpy_array_equal(
     obj: str = ...,
 ): ...
 def assert_extension_array_equal(
-    left,
-    right,
-    check_dtype: bool = ...,
-    check_less_precise: bool = ...,
-    check_exact: bool = ...,
+    left, right, check_dtype: bool = ..., check_exact: bool = ...
```
Comment (Dr-Irv) [inline pandas-stubs/_testing/__init__.pyi:45]:
```suggestion
    left, right, check_dtype: bool = ..., index_values: Index | np.ndarray | None = ..., check_exact: bool = ..., 
    rtol: float = ..., atol: float = ..., obj: str = ...
```

Looking at the docs, the arguments were previously wrong

File: tests/test_plotting.py [hunk 2]
```diff
@@ -486,22 +486,23 @@ def test_plot_scatter(close_figures) -> None:
...
-            IRIS_DF.plot.scatter(x="SepalLength", y="SepalWidth", subplots=True),
-            npt.NDArray[np.object_],
-        ),
-        np.ndarray,
-    )
-    check(
-        assert_type(
-            IRIS_DF.plot(
-                x="SepalLength", y="SepalWidth", subplots=True, kind="scatter"
+    with pytest.warns(UserWarning):
```
Comment (Dr-Irv) [inline tests/test_plotting.py:489]: same issue - not sure why this is there.

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
same issue on `pytest.warns()`.  Also one other thing I saw

File: pandas-stubs/_testing/__init__.pyi (hunk 1 above)
Comment (hamdanal) [reply pandas-stubs/_testing/__init__.pyi]: @Dr-Irv I added all documented parameters.

File: pandas-stubs/_testing/__init__.pyi [hunk 3]
```diff
@@ -6,19 +6,14 @@ from typing import (
...
 )
 
-def assert_almost_equal(
-    left,
-    right,
-    check_dtype: bool | str = ...,
-    check_less_precise: bool | int = ...,
-    **kwargs,
-): ...
+def assert_almost_equal(left, right, check_dtype: bool | str = ..., **kwargs): ...
```
Comment (Dr-Irv) [inline pandas-stubs/_testing/__init__.pyi:16]: From the source, this should be:
```python
def assert_almost_equal(
    left,
    right,
    check_dtype: bool | Literal["equiv"] = ...,
    rtol: float = ...,
    atol: float = ...,
    **kwargs,
) -> None: ...

File: pandas-stubs/_testing/__init__.pyi [hunk 4]
```diff
@@ -51,8 +46,11 @@ def assert_extension_array_equal(
     left,
     right,
     check_dtype: bool = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/_testing/__init__.pyi:48]:
```suggestion
    check_dtype: bool | Literal["equiv"]= ...,
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
I double checked some of the changes vs. the 2.2.2 source, and caught a few more things.

**Comment by hamdanal** [comment]
@Dr-Irv thanks for the review. I went ahead and fixed all the functions in this module.

File: pandas-stubs/_testing/__init__.pyi [hunk 5]
```diff
@@ -1,65 +1,117 @@
...
+    T,
+)
 
 def assert_almost_equal(
-    left,
-    right,
-    check_dtype: bool | str = ...,
-    check_less_precise: bool | int = ...,
+    left: T,
+    right: T,
```
Comment (hamdanal) [inline pandas-stubs/_testing/__init__.pyi:38]: These two parameters could be any object but they have to have the same type. This is represented in the type system by using the same type variable for both parameters .

File: pandas-stubs/_testing/__init__.pyi [hunk 6]
```diff
@@ -103,37 +155,34 @@ def assert_frame_equal(
...
-    check_less_precise: int | bool = ...,
     check_names: bool = ...,
     by_blocks: bool = ...,
     check_exact: bool = ...,
     check_datetimelike_compat: bool = ...,
     check_categorical: bool = ...,
     check_like: bool = ...,
     check_freq: bool = ...,
-    check_flag: bool = ...,
+    check_flags: bool = ...,
```
Comment (hamdanal) [inline pandas-stubs/_testing/__init__.pyi:165]: This was a typo

Comment (Dr-Irv) [reply pandas-stubs/_testing/__init__.pyi:165]: nice catch

**Review by Dr-Irv — APPROVED** [review]
thanks @hamdanal

**Comment by adrian17** [comment]
Hey, we just stumbled upon this PR, as the `pandas.testing` APIs are not typed correctly without it. Are there any plans on getting it merged?

**Comment by Dr-Irv** [comment]
> Hey, we just stumbled upon this PR, as the `pandas.testing` APIs are not typed correctly without it. Are there any plans on getting it merged?

Thanks.  Not sure why I forgot to merge!

**Pull request merged**